src/pages/EventPage.jsx

```diff
@@ -0,0 +1,381 @@
+      setEvents(eventsData.data);
+      console.log(eventsData.data);
+      // setDates();
+      setLocations(getLocation(eventsData.data));
+      setDates(getDate(eventsData.data));
+      const options = { keys: ['name', 'date', 'location'], includeScore: true }; //use date and locaiton selected
+      setFuse(new Fuse(eventsData.data, options));
+    } catch (err) {
+      console.log(`Error getting events: `, err.message);
+    }
+  };
+
+  const getLocation = data => {
+    let location = [];
+    for (let i in data) {
+      location.push(data[i].location);
+    }
+    console.log(location.length);
+    return location;
+  };
+
+  const getDate = data => {
+    let date = [];
+    for (let i in data) {
+      date.push(data[i].date.substring(0, 10));
+    }
+    console.log(date.length);
+    return date;
+  };
+
+  // const getEventId = async () => {
+  //   try {
+  //     const eventIdData = await Backend.get(`/events/${eventId}`);
+  //     console.log(eventIdData);
+  //     setSelectEvent(eventIdData.data);
+  //     console.log(events);
+  //   } catch (err) {
+  //     console.log(`Error getting event ${eventId}: `, err.message);
+  //   }
+  // };
+
+  // const {
+  //   isOpen: isArchiveEventModalOpen,
+  //   onOpen: onArchiveEventModalOpen,
+  //   onClose: onArchiveEventModalClose,
+  // } = useDisclosure();
+  // const confirmArchive = async () => {
+  //   for (const id of selectedEvents) {
+  //     try {
+  //       await Backend.put(`/events/archive/${id}`);
+  //       getEvents();
+  //     } catch (error) {
+  //       console.log(`Error archiving event: ${id}`, error.message);
+  //     }
+  //   }
+  //   onArchiveEventModalClose();
+  //   handleGoBackButton();
+  //   toast({
+  //     title: `Successfully archived ${selectedEvents.length} event(s)`,
+  //     status: 'success',
+  //     duration: 3000,
+  //     isClosable: true,
+  //   });
+  // };
+
+  // const archiveEvents = () => {
+  //   if (selectedEvents.length === 0) handleGoBackButton();
+  //   else onArchiveEventModalOpen();
+  // };
+
+  // const showEvent = () => {
+  //   setShowEvents(true);
+  //   if (eventId) {
+  //     getEventId();
+  //   }
+  // };
+
+  const handleCheckboxChange = id => {
+    const newCheckedItems = [...selectedEvents];
+    const index = newCheckedItems.indexOf(id);
+
+    if (index === -1) {
+      newCheckedItems.push(id);
+    } else {
+      newCheckedItems.splice(index, 1);
+    }
+
+    setSelectedEvents(newCheckedItems);
+    console.log(selectedEvents);
+  };
+
+  const eventCards = displayEvents.map(element => (
+    <GridItem key={element.id}>
+      <EventCard
+        {...element}
+        isSelected={selectedEvents.includes(element.id)}
+        // showSelect={showSelect}
+        handleCheckboxChange={handleCheckboxChange}
+        getEvents={getEvents}
+      />
+    </GridItem>
+  ));
+
+  useEffect(() => {
+    getEvents();
+    console.log(locations);
+    // getEventId(eventId);
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'getEvents' and 'locations'. Either include them or remove the dependency array.